internal/printer/printer.go

```diff
@@ -250,15 +258,19 @@ func (p *printer) printFuncPrelude(opts transform.TransformOptions) {
 	p.hasFuncPrelude = true
 }
 
-func (p *printer) printFuncSuffix(opts transform.TransformOptions) {
+func (p *printer) printFuncSuffix(opts transform.TransformOptions, n *astro.Node) {
 	componentName := getComponentName(opts.Filename)
 	p.addNilSourceMapping()
+	filenameArg := "undefined"
+	propagationArg := "undefined"
 	if len(opts.Filename) > 0 {
 		escapedFilename := strings.ReplaceAll(opts.Filename, "'", "\\'")
-		p.println(fmt.Sprintf("}, '%s');", escapedFilename))
-	} else {
-		p.println("});")
+		filenameArg = fmt.Sprintf("'%s'", escapedFilename)
+	}
+	if n.Transition {
+		propagationArg = "'self'"
 	}
+	p.println(fmt.Sprintf("}, %s, %s);", filenameArg, propagationArg))
```

Comment:
I wonder if the propagation handling could be moved to core somehow 🤔 maybe `renderTransition` could update the propagation type? It would be nice to reduce the params needed to create an Astro component.

Reply:
Hoping to reduce the API surface for the runtime if we ever need another implementation of it, but if it's not possible now I'm also fine with this.